hal/armv8m/exceptions.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * Exception handling
+ *
+ * Copyright 2017, 2022 Phoenix Systems
+ * Author: Pawel Pisarczyk, Jakub Sejdak, Damian Loewnau
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "../exceptions.h"
+#include "../cpu.h"
+#include "../console.h"
+#include "../string.h"
+#include "config.h"
+
+
+void hal_exceptionsDumpContext(char *buff, exc_context_t *ctx, int n)
+{
+	static const char *mnemonics[] = {
+		"0 #InitialSP", "1 #Reset", "2 #NMI", "3 #HardFault",
+		"4 #MemMgtFault", "5 #BusFault", "6 #UsageFault", "7 #SecureFault",
+		"8 #", "9 #", "10 #", "11 #SVC",
+		"12 #Debug", "13 #", "14 #PendSV", "15 #SysTick"
+	};
+	size_t i = 0;
+	u32 msp = (u32)ctx + sizeof(*ctx);
+	u32 psp_s = ctx->psp_s;
+	cpu_hwContext_t *hwctx;
+
+	/* If we came from userspace HW ctx in on psp stack */
+	if (ctx->excret == RET_THREAD_PSP) {
```

Comment:
Not valid on armv8m, causing that userspace exceptions doesn't work correctly. 
I think that you should only check on bit 2 of EXC_RET to determine which stack have hardware context. 

DDI0553B.r D1.2.95 EXC_RETURN, Exception Return Payload:
```
SPSEL, bit [2]
Stack pointer selection. The value of this bit indicates the transitory value of the CONTROL.SPSEL bit
associated with the Security state of the exception as indicated by EXC_RETURN.ES.
The possible values of this bit are:
0
Main stack pointer.
1
Process stack pointer.

```

It's differ from armv7m mainly because of additional security features (trustzone etc.) I would advise to check other usage of that defines because it indicates secure mode

Reply:
After discussion we have conclusions:
- on nrf9160 default execution state is secure state, so EXC_RET is set correctly
- it was still worth to switch to checking on bit 2  in case of exceptions - it's well defined in cpu spec. 

So current version is ok for me and it's proved to work.